src/main/java/com/softserve/academy/spaced/repetition/service/CourseService.java

```diff
@@ -126,6 +126,13 @@
      */
     void deleteLocalCourse(Long courseId) throws NotAuthorisedUserException;
 
+    /**
+     * Deletes course from DB by Admin
+     *
+     * @param courseId must not be {@literal null}.
+     */
+    void deleteCourseByAdmin (Long courseId);
```

Comment:
It is not the service responsibility to decide is the admin deleting the course. Therefore (optionally) I would rename it to something like "deleteCourseAndSubscriptions" to really reflect what the method do, not who is the caller of the method.